pkg/upgrade/control_plane.go

```diff
@@ -29,9 +29,10 @@ import (
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
-	bootstrapv1 "sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/api/v1alpha2"
-	kubeadmv1beta1 "sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/kubeadm/v1beta1"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha2"
+	clusterv1a3 "sigs.k8s.io/cluster-api/api/v1alpha3"
```

Comment:
This tool is only for v1alpha2 and it is not applicable for v1alpha3 clusters (upgrade functionality is being built in to v1alpha3). The path forward to fix the issue is to make a pull request to the deprecated cluster-api-bootstrap-provider-kubeadm repository, adjusting the `omitempty` tags in the kubeadm types. Once we merge that pull request and tag a new release, you can update this one to pull in the the new tag.

Reply:
Sure. Will do that.

Reply:
We have a new CABPK release, so you should be able to proceed now. Thanks!